feat: 멤버 클래스, 테이블 구현

ShinHeeEul opened the pull request:
## 개요
- [x] JPA 환경 설정(application.yml파일 작성)
- [x] Member 테스트 코드 작성
- [x] Member 클래스 코드 작성

## 작업사항
- member 테이블 생성시 쿼리 7번 발생

## 주의사항
- 실행 환경 변수, 테스트 환경 변수를 설정해주어야 정상 작동

ShinHeeEul committed:
docs: todoList 작성

ShinHeeEul committed:
feat: Member 필드 정규 표현식 상수로 표현

ShinHeeEul committed:
fix: TODO 파일 수정

ShinHeeEul committed:
fix: 닉네임 정규 표현식 수정

ShinHeeEul committed:
feat: 멤버 테스트 파일 작성

ShinHeeEul committed:
feat: Lombok, hibernate.validator 추가

ShinHeeEul committed:
chore: 환경 변수 설정

ShinHeeEul committed:
feat: Member 클래스 구현

jung-woo-kim commented on src/main/java/bangwool/server/TODO.md:
TODO.md 파일은 지우고 pr 보내주세요.

jung-woo-kim commented on src/main/java/bangwool/server/domain/Regex.java:
이 정규 표현식은 Member 엔티티에서만 사용될 것 같은데 Member 엔티티 내부로 옮기는 것은 어떤가요?

jung-woo-kim commented on src/main/resources/application.yml:
```diff
@@ -0,0 +1,132 @@
+  sql:
+    init:
+      platform: mysql
+
```
localDB 는 보통 h2에서 실행해서 h2로 바꿔주시면 좋을 것 같아요

jung-woo-kim commented on src/main/resources/application.yml:
```diff
@@ -0,0 +1,132 @@
+        naming:
+          physical-strategy: org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl
+    generate-ddl: false
+
```
jpa의 ddl-auto create기능은 local에서만 사용하도록 해야합니다.
보통 local : create /dev : update / prod : validate 이런식으로 사용해요

jung-woo-kim commented on src/test/java/bangwool/server/domain/MemberTest.java:
```diff
@@ -0,0 +1,120 @@
+        }
+    }
+
+}
```
eof 추가 부탁드립니다.

jung-woo-kim commented on src/test/java/bangwool/server/domain/MemberTest.java:
멤버 단위 테스트 구현 잘해주셨네요 👍

jung-woo-kim reviewed (changes requested):
구현 확인했습니다! 리뷰 담긴 내용들 확인하시고 변경 사항 확인되면 approve 할게요!

ShinHeeEul committed:
fix: 코드 리뷰 반영

- Regex in Member
- local h2로 변경
- ddl-auto local,dev,prod에 맞게 설정
- eol추가
- TODO 파일 삭제
